src/MessageHandler.ts

```diff
@@ -1,90 +1,81 @@
 import Discord from 'discord.js';
 import './discord/Message';
 
-import QueueItem from './queue/QueueItem';
+import * as Commands from './commands/Commands';
+
+import DatabaseAdapter from './db/DatabaseAdapter';
 import SoundQueue from './queue/SoundQueue';
-import Util from './Util';
 
 export default class MessageHandler {
-  private readonly prefix: string;
   private readonly queue: SoundQueue;
+  private readonly db: DatabaseAdapter;
+  private readonly prefix: string;
 
-  constructor(queue: SoundQueue, prefix: string) {
+  constructor(queue: SoundQueue, db: DatabaseAdapter, prefix: string) {
     this.queue = queue;
+    this.db = db;
     this.prefix = prefix;
   }
 
   public handle(message: Discord.Message) {
     if (message.isDirectMessage()) return;
     if (!message.hasPrefix(this.prefix)) return;
-    if (Util.isIgnoredUser(message.author)) return;
+
+    // @REVIEW Move this check to User.isIgnored?
+    // Then user knows about db :/
+    if (this.db.isIgnoredUser(message.author.id)) return;
 
     message.content = message.content.substring(this.prefix.length);
     this.handleMessage(message);
   }
 
+  // @REVIEW Want to extract this to a CommandHandler class that maps trigger -> Command
```

Comment:
CommandHandler can take db and queue as constructor arguments. Input and message are passed then to its `handleMessage` method. What does the ignore parameter do?

Reply:
What I meant here is: Right now, the MessageHandler knows exactly which trigger belongs to which Command and which parameters are required. This is too much hardcoded, isn't it? So instead, I want a field `trigger` on each command that contains the triggers (e.g. ['stop', 'leave']). Then I just instantiate the command that the trigger belongs to (e.g. StopCommand) and pass all of those parameters to that command, even though it might just need one of them (e.g. db for StopCommand), because we the CommandHandler does not know anymore which command needs which parameters. BaseCommand just takes a variable number of args. Is that how it should be done?